corbench/pkg/provider/k8s/k8s.go

```diff
@@ -0,0 +1,1557 @@
+// Copyright 2019 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8s
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"gopkg.in/alecthomas/kingpin.v2"
+	admissionregistration "k8s.io/api/admissionregistration/v1"
+	appsV1 "k8s.io/api/apps/v1"
+	batchV1 "k8s.io/api/batch/v1"
+	apiCoreV1 "k8s.io/api/core/v1"
+	apiNetworkingV1 "k8s.io/api/networking/v1"
+	rbac "k8s.io/api/rbac/v1"
+	apiServerExtensionsV1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	apiServerExtensionsClient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	apiMetaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	"k8s.io/client-go/util/retry"
+	_ "k8s.io/cloud-provider-gcp/pkg/clientauthplugin/gcp"
+
+	"github.com/cortexproject/test-infra/corbench/pkg/provider"
+)
+
+func init() {
```

Comment:
ditto. If we cannot reuse prombench CLI, let's try to use its library instead of copy its code directly.